src/interpreter/ByteCodeGenerator.cpp

```diff
@@ -88,14 +88,34 @@ void ByteCodeGenerateContext::morphJumpPositionIntoComplexCase(ByteCodeBlock* cb
 }
 
 #ifdef ESCARGOT_DEBUGGER
-void ByteCodeGenerateContext::insertBreakpoint(size_t line, Node* node)
+void ByteCodeGenerateContext::insertBreakpoint(size_t index, Node* node)
 {
     ASSERT(m_breakpointContext != nullptr);
+    ASSERT(index != SIZE_MAX);
+
+    StringView src = m_codeBlock->asInterpretedCodeBlock()->src();
+    ExtendedNodeLOC sourceElementStart = m_codeBlock->asInterpretedCodeBlock()->functionStart();
+    size_t lastLineOffset = m_breakpointContext->m_lastBreakpointLineOffset;
+    index -= sourceElementStart.index;
+
+    if (index > m_breakpointContext->m_lastBreakpointIndexOffset) {
```

Comment:
IMO index would be increased sequentially, so this check code seems always true.